src/main/java/com/google/plantasy/servlets/getUserImagesForQuizPage.java

```diff
@@ -39,15 +40,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         HashMap<String,String> user_ids_and_pictures = new HashMap<String, String>();
 
         Entity current_game = UserUtils.getEntityFromDatastore("Game", "gameId", (userEntity.getProperty("gameId")).toString(), datastore);
-        for(String player : (ArrayList<String>) current_game.getProperty("userIds")) {
-            if(!userEntity.getProperty("userID").equals(player)) {
-                Entity playerEntity = UserUtils.getEntityFromDatastore("user","userID", player, datastore);
-                user_ids_and_pictures.put(player, (playerEntity.getProperty("blobKey")).toString());
+        for(String player_name : (ArrayList<String>) current_game.getProperty("userIds")) {
+            if(!userEntity.getProperty("userID").equals(player_name)) {
+                User player = new User(UserUtils.getEntityFromDatastore("user","userID", player_name, datastore));
+                try {
+                    user_ids_and_pictures.put(player.getName(), player.getBlobKey());
+                } catch (Exception e){
+                    // If player.getBlobKey() causes a NPE, we can skip that user.
```

Comment:
IIUC `player.getBlobKey()` will return `null` rather than throw a NPE. And adding a null value to a HashMap is also legal. So instead I think we would need to check for a null blob key in the `if` condition.